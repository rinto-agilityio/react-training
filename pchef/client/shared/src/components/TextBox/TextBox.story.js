/* eslint no-console: "off" */
// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Components
import TextBox from '.'

// Helper
import { wInfo } from '../../../.storybook/utils';

const refInput = {
  current: null,
}

storiesOf('TextBox', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <TextBox
      placeholder="Default"
      refInput={refInput}
      onBlur={action(value => value.target.value)}
      onChangeText={action(value => value)}
    />
  ))
  .add('Multiline', () => (
    <TextBox
      placeholder="Multiline"
      multiline
      refInput={refInput}
      onBlur={action(value => value)}
      onChangeText={action(value => value)}
    />
  ))
