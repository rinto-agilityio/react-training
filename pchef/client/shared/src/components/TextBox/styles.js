// Libs
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../themes'

const styles = StyleSheet.create({
  default: {
    height: METRICS.input.md,
    borderColor: COLORS.grayDarker,
    borderBottomWidth: 1,
    fontSize: FONTS.fontSize.medium,
    color: COLORS.black,
  },
})

export default styles
