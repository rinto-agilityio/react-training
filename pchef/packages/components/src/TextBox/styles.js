// Libs
import { StyleSheet } from 'react-native';

// Themes
import { COLORS } from 'themes/Colors';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1
  },
  default: {
    height: 40,
    borderBottomColor: COLORS.inputBorder,
    borderBottomWidth: 1,
    fontSize: 18,
    flex: 1,
    color: COLORS.black
  },
});