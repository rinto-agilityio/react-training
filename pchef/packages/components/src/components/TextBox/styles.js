// Libs
import { StyleSheet } from 'react-native';

// Themes
import { COLORS } from '../../themes';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1
  },
  default: {
    height: 40,
    borderColor: COLORS.inputBorder,
    borderBottomWidth: 1,
    fontSize: 18,
    flex: 1,
    color: COLORS.black
  },
});