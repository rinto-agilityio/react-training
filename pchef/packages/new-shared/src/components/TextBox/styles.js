// Libs
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../themes'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
  },
  default: {
    height: METRICS.mediumInput,
    borderColor: COLORS.grayDarker,
    borderBottomWidth: 1,
    fontSize: FONTS.fontSize.medium,
    color: COLORS.black,
    flex: 1,
  },
})

export default styles
