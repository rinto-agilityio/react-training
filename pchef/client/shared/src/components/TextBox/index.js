// @flow
// add flow above to fix for using flow with React.memo

// Libs
import React, { memo } from 'react'
import { TextInput, View } from 'react-native'

// Styles
import styles from './styles'

// Themes
import { COLORS } from '../../themes'

type Props = {
  multiline?: boolean,
  autoCapitalize?: string,
  autoCorrect?: boolean,
  customStyle?: {} | Array<{}>,
  editable?: boolean,
  placeholder?: string,
  placeholderTextColor?: string,
  numberOfLines?: number,
  defaultValue?: string,
  refInput: {
    current: {
      _node: HTMLInputElement,
      _lastNativeText: string,
    } |
    null
  },
  onSubmitEditing?: () => void,
  secureTextEntry?: boolean,
  customContainer?: {} | Array<{}>,
}

const TextBox = ({
  multiline,
  autoCapitalize,
  autoCorrect,
  customStyle,
  editable,
  placeholder,
  placeholderTextColor,
  numberOfLines,
  defaultValue,
  refInput,
  onSubmitEditing,
  secureTextEntry,
  customContainer,
}: Props) => (
  <View style={[styles.container, customContainer]}>
    <TextInput
      multiline={multiline}
      autoCapitalize={autoCapitalize}
      autoCorrect={autoCorrect}
      defaultValue={defaultValue}
      placeholder={placeholder}
      editable={editable}
      numberOfLines={numberOfLines}
      style={[styles.default, customStyle]}
      ref={refInput}
      placeholderTextColor={placeholderTextColor}
      onSubmitEditing={onSubmitEditing}
      secureTextEntry={secureTextEntry}
    />
  </View>
)

TextBox.defaultProps = {
  multiline: false,
  autoCapitalize: 'none',
  autoCorrect: false,
  editable: true,
  numberOfLines: 1,
  placeholder: '',
  customStyle: {},
  placeholderTextColor: COLORS.black,
  onSubmitEditing: () => {},
  defaultValue: '',
  secureTextEntry: false,
  customContainer: {},
}

export default memo<Props>(TextBox)
