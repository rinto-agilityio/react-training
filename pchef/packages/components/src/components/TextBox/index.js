/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

// Libs
import React from 'react';
import { TextInput, View } from 'react-native';

// Styles
import { styles } from './styles';

// Themes
import { COLORS } from '../../themes';

const TextBox = ({
  multiline,
  autoCapitalize,
  autoCorrect,
  customStyle,
  editable,
  placeholder,
  placeholderTextColor,
  numberOfLines,
  defaultValue,
  refInput,
  onSubmitEditing
}) => (
    <View style={styles.container}>
      <TextInput
        multiline={multiline}
        autoCapitalize={autoCapitalize}
        autoCorrect={autoCorrect}
        defaultValue={defaultValue}
        placeholder={placeholder}
        multiline={multiline}
        editable={editable}
        numberOfLines={numberOfLines}
        style={[styles.default, customStyle]}
        ref={refInput}
        placeholderTextColor={placeholderTextColor}
        onSubmitEditing={onSubmitEditing}
      />
    </View>
  )

TextBox.defaultProps = {
  multiline: false,
  autoCapitalize: 'none',
  autoCorrect: false,
  editable: true,
  numberOfLines: 1,
  placeholder: '',
  customStyle: {},
  placeholderTextColor: COLORS.black
}

export default TextBox