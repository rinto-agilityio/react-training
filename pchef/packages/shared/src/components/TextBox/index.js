// Libs
import React from 'react'
import { TextInput, View } from 'react-native'

// Styles
import { styles } from './styles'

// Themes
import { COLORS } from '../../themes'

type Props = {
  multiline?: boolean,
  autoCapitalize?: string,
  autoCorrect?: boolean,
  customStyle?: {},
  editable?: boolean,
  placeholder?: string,
  placeholderTextColor?: string,
  numberOfLines?: number,
  defaultValue?: string,
  refInput: { current: HTMLInputElement | null },
  onSubmitEditing?: () => void,
}

const TextBox = ({
  multiline,
  autoCapitalize,
  autoCorrect,
  customStyle,
  editable,
  placeholder,
  placeholderTextColor,
  numberOfLines,
  defaultValue,
  refInput,
  onSubmitEditing,
}: Props) => (
  <View style={styles.container}>
    <TextInput
      multiline={multiline}
      autoCapitalize={autoCapitalize}
      autoCorrect={autoCorrect}
      defaultValue={defaultValue}
      placeholder={placeholder}
      editable={editable}
      numberOfLines={numberOfLines}
      style={[styles.default, customStyle]}
      ref={refInput}
      placeholderTextColor={placeholderTextColor}
      onSubmitEditing={onSubmitEditing}
    />
  </View>
)

TextBox.defaultProps = {
  multiline: false,
  autoCapitalize: 'none',
  autoCorrect: false,
  editable: true,
  numberOfLines: 1,
  placeholder: '',
  customStyle: {},
  placeholderTextColor: COLORS.black,
  onSubmitEditing: () => {},
  defaultValue: '',
}

export default TextBox
