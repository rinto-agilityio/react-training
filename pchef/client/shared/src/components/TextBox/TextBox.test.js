// Libs
import renderer from 'react-test-renderer'

// Components
import { TextInput } from 'react-native'
import TextBox from '.'

const props = {
  onBlur: jest.fn(),
  onChangeText: jest.fn(),
}

const component = shallow(<TextBox {...props} />)

describe('TextBox', () => {
  it('Renders correctly', () => {
    const textbox = renderer
      .create(<TextBox placeholder="Multiline" multiline />)
      .toJSON()
    expect(textbox).toMatchSnapshot()
  })

  it('Should call onBlur when blur on text input', () => {
    const textbox = component.find(TextInput)
    textbox.simulate('blur', {
      target: { value: 'hello' },
    })
    expect(props.onBlur).toHaveBeenCalled()
  })

  it('Should call onChangeText when input text', () => {
    const textbox = component.find(TextInput)
    textbox.props().onChangeText()
    expect(props.onChangeText).toHaveBeenCalled()
  })
})
