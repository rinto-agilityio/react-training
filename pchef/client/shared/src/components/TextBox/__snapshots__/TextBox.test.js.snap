// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <textarea
    autoCapitalize="none"
    autoComplete="on"
    autoCorrect="off"
    className="css-textinput-1cwyjr8 r-borderBottomWidth-qklmqi r-borderColor-34efmp r-color-cqee49 r-fontSize-1i10wst r-height-eu3ka r-placeholderTextColor-whyam9"
    data-focusable={true}
    defaultValue=""
    dir="auto"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onSelect={[Function]}
    placeholder="Multiline"
    readOnly={false}
    rows={1}
    spellCheck={false}
  />
</div>
`;
